animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -132,11 +140,29 @@ cdef class Mesh:
         return list(self.vertexColorLayers.items())
 
     def getVertexColorLayerNames(self):
-        return list(self.vertexColorLayers.keys())    
-    
+        return list(self.vertexColorLayers.keys())
+
     def getVertexColors(self, str colorLayerName):
         return self.vertexColorLayers.get(colorLayerName, None)
-        
+
+    def getVertexLinkedVertices(self, long vertexIndex):
+        cdef IntegerList neighboursAmounts, neighboursStarts, neighbours, neighbourEdges
+        neighboursAmounts, neighboursStarts, neighbours, neighbourEdges = self.getLinkedVerticesCache()
+
+        cdef int start, amount
+        amount = neighboursAmounts.data[vertexIndex]
+        start = neighboursStarts.data[vertexIndex]
+
+        cdef IntegerList vertexNeighbours = IntegerList(length = amount)
+        cdef IntegerList vertexNeighbourEdges = IntegerList(length = amount)
+
+        cdef int i, index
+        for i in range(amount):
+            index = i + start
+            vertexNeighbours.data[i] = neighbours.data[index]
+            vertexNeighbourEdges.data[i] = neighbourEdges.data[index]
```

Comment:
I wounder if we can use `memcpy` here.
Import it `from libc.string cimport memcpy`.
Then check its man page in section 3 to see how to use it.

Reply:
Okay.

Reply:
Omar, I have tried that and it is working. 

How can I slice the `InterList` e.g., `neighbours.data[start : end]`?

Reply:
Why would you need to slice the list?

Reply:
The `neighbours` is a list of neighbours for all vertices. To get neighbours for a vertex we need to slice this list. It is happening in the loop from `start - start + amount`: https://github.com/JacquesLucke/animation_nodes/blob/9b1b9cce710d263cf95dc5fc67953926f1241dfb/animation_nodes/data_structures/meshes/mesh_data.pyx#L160-L163

Reply:
Yes, that's why I am proposing to use `memcpy` to copy from the `neighbours` lists to the output lists. What is your current code?

Reply:
I think slicing should work as well. Internally it will do the same as memcpy I think, just with a little bit more overhead. Use `neighbours[start:end]` if I remember correctly.

Reply:
I thought we are not going to use for loop. I run this code just for testing it for a list
```
cdef IntegerList vertexNeighboursTest = IntegerList(length = amount)
memcpy(vertexNeighboursTest.data, vertexNeighbours.data, sizeof(vertexNeighbours))
```

Reply:
I looked into the `getValuesInSlice` code, it does a bit more than we need. While it has little overhead, explicit memcpy is just two extra lines of code. So I think it is acceptable.

Reply:
But in a loop it gives a error for this code,
```
for i in range(amount):
    index = i + start
    memcpy(vertexNeighbours.data[i], neighbours.data[index], sizeof(neighbours.data[index]))
```
![Screenshot from 2019-12-31 17-25-54](https://user-images.githubusercontent.com/30294746/71620973-ae9fa700-2bf2-11ea-93d9-cea069978025.png)

Reply:
I meant something like this:
```cython
memcpy(vertexNeighbours.data, neighbours.data + start, amount * sizeof(long));
memcpy(vertexNeighbourEdges.data, neighbourEdges.data + start, amount * sizeof(long));
```

Reply:
Okay

Reply:
Omar, we run the animation, the Output is flickering. Is it due to _[To avoid overflows, the size of the arrays pointed to by both the destination and source parameters, shall be at least num bytes, and should not overlap (for overlapping memory blocks, memmove is a safer approach).](http://www.cplusplus.com/reference/cstring/memcpy/)_

Reply:
Ah, probably because we are using integer lists while we have `sizeof(long)`. It should be noted that Integer lists in AN are actually Long lists behind the scenes, so you should return a Long list instead. In conclusion, use LongList instead.

Since we already encountered an unexpected behavior, we should probably listen to @JacquesLucke, use slicing for safety and readability. :smile:

Reply:
I have made the changes. It is working fine with the slicing.